File: algo_2021/s7.-hash-heap/128.Hash_Function/128_hash_function.java
public class Solution {
    /**
     * @param key: A string you should hash
     * @param HASH_SIZE: An integer
     * @return: An integer
     */
    public int hashCode(char[] key, int HASH_SIZE) {
        // write your code here
        long ans = 0;
        for (int i = 0; i < key.length; i++) {
            ans = (ans * 33 + (int)key[i]) % HASH_SIZE; 
        }

        return (int)ans;
        
    }
}

File: lintcode_algo_2021/s7.-ds-design/128.Hash_Function/128_hash_function.py
class Solution:
    """
    @param key: A string you should hash
    @param HASH_SIZE: An integer
    @return: An integer
    """
    def hashCode(self, key, HASH_SIZE):
        # write your code here
        hash_value = 0
        for c in key: 
            hash_value = (hash_value * 33 + ord(c)) % HASH_SIZE
        return hash_value